### ISSUE TITLE
[en] False positive comma suggestion before "so"

### ISSUE THREAD

**Issue by donnerpeter** [languagetool-org/languagetool#5618]
Here:
`Consider rewriting the method so it becomes easier to understand.`

A linguist tells me the comma isn't necessary.

**Comment by AzadehSafakish** [comment]
Yes, a comma is not necessary in the example sentence and your generalization is correct.

**Comment by qenya** [comment]
I just encountered this false positive too. In fact, a lot of the example sentences make the same mistake!

For example:
> I need to borrow some money so I can buy some medicine for Tom.

This means "I need to borrow some money in order to buy some medicine for Tom."

However, the rule description suggests "correcting" it to:
> I need to borrow some money, so I can buy some medicine for Tom.

This means "I can buy some medicine for Tom because I need to borrow some money", which is clearly not correct.
